src/Simulation/EntryPointDriver/Driver.cs

```diff
@@ -241,146 +316,241 @@ private void AddOptionIfAvailable<T>(Command command, OptionInfo<T> option)
+            }
         }
-    }
 
-    /// <summary>
-    /// A modification of the command-line <see cref="HelpBuilder"/> class.
-    /// </summary>
-    internal sealed class QsHelpBuilder : HelpBuilder
-    {
         /// <summary>
-        /// Creates a new help builder using the given console.
+        /// Struct for housing a command with its validators.
         /// </summary>
-        /// <param name="console">The console to use.</param>
-        internal QsHelpBuilder(IConsole console) : base(console)
+        private struct CommandWithValidators
         {
-        }
+            public Command Command;
+            public IEnumerable<ValidateSymbol<CommandResult>> Validators;
 
-        protected override string ArgumentDescriptor(IArgument argument)
-        {
-            // Hide long argument descriptors.
-            var descriptor = base.ArgumentDescriptor(argument);
-            return descriptor.Length > 30 ? argument.Name : descriptor;
+            /// <summary>
+            /// Basic constructor.
+            /// </summary>
+            public CommandWithValidators(Command command, IEnumerable<ValidateSymbol<CommandResult>> validators)
+            {
+                Command = command;
+                Validators = validators;
+            }
+
+            /// <inheritdoc/>
+            public override bool Equals(object? obj)
+            {
+                return obj is CommandWithValidators other &&
+                       EqualityComparer<Command>.Default.Equals(Command, other.Command) &&
+                       EqualityComparer<IEnumerable<ValidateSymbol<CommandResult>>>.Default.Equals(Validators, other.Validators);
+            }
+
+            /// <inheritdoc/>
+            public override int GetHashCode()
+            {
+                return HashCode.Combine(Command, Validators);
+            }
+            
+            /// <summary>
+            /// Allows the type to be deconstructed as a tuple.
+            /// </summary>
+            public void Deconstruct(out Command command, out IEnumerable<ValidateSymbol<CommandResult>> validators)
+            {
+                command = Command;
+                validators = Validators;
+            }
+
+            /// <summary>
+            /// Allows the type to be converted to a tuple.
+            /// </summary>
+            public static implicit operator (Command Command, IEnumerable<ValidateSymbol<CommandResult>> Validators)(CommandWithValidators value)
+            {
+                return (value.Command, value.Validators);
+            }
+
+            /// <summary>
+            /// Allows the type to be converted from a tuple.
+            /// </summary>
+            public static implicit operator CommandWithValidators((Command Command, IEnumerable<ValidateSymbol<CommandResult>> Validators) value)
+            {
+                return new CommandWithValidators(value.Command, value.Validators);
+            }
```

Comment:
This is a lot of code for such a small type, and the implicit conversions to and from the tuple type don't seem necessary. Wouldn't something like

```csharp
private struct CommandWithValidators
{
    public Command Command;
    public IReadOnlyCollection<ValidateSymbol<CommandResult>> Validators;
}
```

work? (Maybe with a constructor as well.)

Reply:
I just used the the intellisense suggestion to generate the equivalent type for the tuple. The tuple conversions make it easy to do the deconstructing.

Reply:
The deconstructing is done by the `Deconstruct` method, not the tuple conversions.

I think the amount of code should be proportional to the scale of the problem. The problem being solved here is needing a named container with two fields. I think that should be a few lines of code (1 line of code in C# 9 if you use a positional record), rather than 50.

I also think that implicit conversions should be used sparingly, and being able to write `(command, validators)` instead of `new CommandWithValidators(command, validators)` does not seem like it justifies adding implicit conversion operators.

Reply:
Simplified to the struct provided, with a constructor.